### PR TITLE
[FAD-124] - Fix connect wallet button

### PR DESCRIPTION
**Pull request by luansievers**
### Description

> Wallet connect button was not working properly after the goldfinch update.
### Checklist

- [ ] `CHANGELOG` has been updated (_if appropriate_)
- [ ] Environment variables updated (_if appropriate_)

### CODE REVIEW

File: packages/client3/components/design-system/wallet-button/wallet-button.tsx [hunk 1]
```diff
@@ -1,151 +1,61 @@
...
-    <Tooltip content={buttonStyles.tooltip}>
+  return error ? (
+    <Tooltip content={error.message}>
       <Button
-        onClick={handleWalletConnect}
-        isLoading={{ isLoading, position: "left" }}
-        iconLeft={buttonStyles.icon as IconProps["name"]}
-        buttonType={buttonStyles.type as ButtonType}
+        iconLeft="Exclamation"
+        buttonType={ButtonType.PRIMARY as ButtonType}
```
Comment (v1n33th) [inline packages/client3/components/design-system/wallet-button/wallet-button.tsx:26]: Is type casting required? If not can we add the type to the Button component

**Review by v1n33th — APPROVED** [review]
Just a minor suggestion

File: packages/client3/components/design-system/wallet-button/wallet-button.tsx (hunk 1 above)
Comment (luansievers) [reply packages/client3/components/design-system/wallet-button/wallet-button.tsx]: its not, I removed the casting

File: packages/client3/components/design-system/wallet-button/types/wallet-button.types.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-  state: string;
-  tooltip?: string;
-}
+// import { ButtonType } from "@/components/design-system";
+// export interface IWalletButtonStyles {
+//   icon: string;
+//   type: ButtonType;
+//   state: string;
+//   tooltip?: string;
+// }
```
Comment (luansievers) [inline packages/client3/components/design-system/wallet-button/types/wallet-button.types.ts:8]:
```suggestion
```

**Review by v1n33th — APPROVED** [review]

File: packages/client3/components/design-system/wallet-button/wallet-button.tsx [hunk 3]
```diff
@@ -1,151 +1,96 @@
...
+  "onClick" | "isLoading" | "iconLeft" | "buttonType" | "className" | "children"
+>;
+
 export function WalletButton({ className }: Props) {
-  const { account, isActive, error } = useWallet();
-  const [isLoading, setIsLoading] = useState(false);
+  const { account } = useWallet();
+
+  const isActivating = metaMaskHooks.useIsActivating();
+  const error = metaMaskHooks.useError();
```
[the remark this answers is not in the document]
Comment (luansievers) [reply packages/client3/components/design-system/wallet-button/wallet-button.tsx:38]: These are different, I got from the new code from goldfinch...

**Review by sachasmart-weavik — APPROVED** [review]
https://user-images.githubusercontent.com/98773781/196442713-a092ac7e-bc4a-4d49-b79d-4194af735f96.mp4

**Pull request merged**